From 51a7e0ff51a64046356fcbb460331a3a8178e82e Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Tue, 15 Apr 2014 16:38:31 +0100
Subject: [PATCH 194/351] regulator: arizona-ldo1: Sync with upstream comments

Use of IS_ENABLED is preferred for configuration specific code, so use
that instead of of ifdef.

Change-Id: I1572ffafaf0657a6356d30aeb6f0de13286452e9
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/regulator/arizona-ldo1.c |   30 +++++++-----------------------
 1 file changed, 7 insertions(+), 23 deletions(-)

diff --git a/drivers/regulator/arizona-ldo1.c b/drivers/regulator/arizona-ldo1.c
index 7f1660c..daccca6 100644
--- a/drivers/regulator/arizona-ldo1.c
+++ b/drivers/regulator/arizona-ldo1.c
@@ -177,7 +177,6 @@ static const struct regulator_init_data arizona_ldo1_default = {
 	.num_consumer_supplies = 1,
 };
 
-#ifdef CONFIG_OF
 static int arizona_ldo1_of_get_pdata(struct arizona *arizona,
 				     struct regulator_config *config)
 {
@@ -214,22 +213,6 @@ static int arizona_ldo1_of_get_pdata(struct arizona *arizona,
 	return 0;
 }
 
-static void arizona_ldo1_of_put_pdata(struct regulator_config *config)
-{
-	of_node_put(config->of_node);
-}
-#else
-static inline int arizona_ldo1_of_get_pdata(struct arizona *arizona,
-					    struct regulator_config *config)
-{
-	return 0;
-}
-
-static inline void arizona_ldo1_of_put_pdata(struct regulator_config *config)
-{
-}
-#endif
-
 static int arizona_ldo1_probe(struct platform_device *pdev)
 {
 	struct arizona *arizona = dev_get_drvdata(pdev->dev.parent);
@@ -273,10 +256,12 @@ static int arizona_ldo1_probe(struct platform_device *pdev)
 	config.driver_data = ldo1;
 	config.regmap = arizona->regmap;
 
-	if (!dev_get_platdata(arizona->dev)) {
-		ret = arizona_ldo1_of_get_pdata(arizona, &config);
-		if (ret < 0)
-			return ret;
+	if (IS_ENABLED(CONFIG_OF)) {
+		if (!dev_get_platdata(arizona->dev)) {
+			ret = arizona_ldo1_of_get_pdata(arizona, &config);
+			if (ret < 0)
+				return ret;
+		}
 	}
 
 	config.ena_gpio = arizona->pdata.ldoena;
@@ -301,8 +286,7 @@ static int arizona_ldo1_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	if (!dev_get_platdata(arizona->dev))
-		arizona_ldo1_of_put_pdata(&config);
+	of_node_put(config.of_node);
 
 	platform_set_drvdata(pdev, ldo1);
 
-- 
1.7.9.5

